psutil/_pslinux.py

```diff
@@ -1345,6 +1345,18 @@ def sensors_temperatures():
             current = float(bcat(path)) / 1000.0
             path = os.path.join(os.path.dirname(base), 'name')
             unit_name = cat(path).strip()
+
+            address = os.path.join(os.path.dirname(base), 'device', 'address')
```

Comment:
What's the content of this file? Can you paste it here? The parsing logic that you apply later on:
```python
unit_name += '-' + ''.join(cat(address).strip().split(':')[1:3]).split('.')[0]
```
...looks a bit convoluted, so I'd like to simplify it if possible.

Reply:
Of course, I have two files:
- `/sys/class/hwmon/hwmon0/device/address` containing
```0000:b3:00.0```
- `/sys/class/hwmon/hwmon1/device/address` containing
```0000:b4:00.0```

These correspond to the PCIe locations of my SSDs (more about pcie addressing here https://dassencio.org/75)
```
domain:bus:device.function
0000  :b3 :00    .0
```

The line you asked me about:
1. takes the content of the file (stripping any white-spaces)
```python
cat(address).strip()
# '0000:b3:00.0'
```
2. splits the content at each `:`
```python
cat(address).strip().split(':')
# ['0000', 'b3', '00.0']
```
3. discards the domain number (quite always zero), takes the second and third part of the address (bus, device and function number)
```python
cat(address).strip().split(':')[1:3]
# ['b3', '00.0']
```
4. joins them together
```python
''.join(cat(address).strip().split(':')[1:3])
# 'b300.0'
```
5. splits the string at the `.` and then takes only the first part (function number is not needed since SSDs will only have one function, ones with more are not likely to have more sensors on different functions)
```python
''.join(cat(address).strip().split(':')[1:3]).split('.')[0]
# 'b300'
```
6. finally appends the address to the unit name separated by a `-`
```python
unit_name += '-' + ''.join(cat(address).strip().split(':')[1:3]).split('.')[0]
```